src/routing.ts

```diff
@@ -301,13 +301,38 @@ export function createRouterContext(
   }
 
   const [isRouting, setIsRouting] = createSignal(false);
-  const start = async (callback: () => void) => {
-    setIsRouting(true);
-    try {
-      await startTransition(callback);
-    } finally {
-      setIsRouting(false);
-    }
+  let activeTransitions = 0;
+
+  // Keep track of last intent and target, so that last call to transition wins
+  let lastTransitionIntent: Intent | undefined;
+  let lastTransitionTarget: LocationChange | undefined;
+
+  // Transition the location to a new value
+  const transition = (newIntent: Intent, newTarget: LocationChange) => {
+    intent = lastTransitionIntent = newIntent;
+    lastTransitionTarget = newTarget;
+
+    if (++activeTransitions === 1) setIsRouting(true);
```

Comment:
Can `activeTransition` be replaced by comparing `newTarget` and `lastTransitionTarget`? [In my own testing](https://github.com/jpdutoit/solid-router/compare/is-routing-and-intent...Brendonovich:solid-router:442-improvements) it seems to work

Reply:
Yeah, this is cleaner! I pulled in your commits